src/nvim/api/buffer.c

```diff
@@ -193,12 +193,14 @@ void buffer_set_line_slice(Buffer buffer,
   buf_T *save_curbuf = NULL;
   win_T *save_curwin = NULL;
   tabpage_T *save_curtab = NULL;
-  size_t new_len = replacement.size;
-  size_t old_len = (size_t)(end - start);
-  ssize_t extra = 0;  // lines added to text, can be negative
-  char **lines = (new_len != 0) ? xcalloc(new_len, sizeof(char *)) : NULL;
-
-  for (size_t i = 0; i < new_len; i++) {
```

Comment:
Sorry, but I don't agree to all this changing size_t to other types.
All these had been previously refactored to `size_t`, as it is the natural type for sizes, lenghts, counts, indexes, etc.
Changing all of them just because at some point they interact with a `linenr_T` is not what desired.
Better to (safely) cast the involved `linenr_T` when needed.
I write this here cause it's the first occurrence, but what I said applies to many other changes after this.